fmriprep/workflows/bold.py

```diff
@@ -317,112 +346,29 @@ def init_func_preproc_wf(bold_file, ignore, dismiss_t1w, freesurfer,
-                            'nonlinear susceptibility correction for dataset %s.', bold_file)
-                summary.inputs.distortion_correction = 'SyN'
-                workflow.connect([
-                    (nonlinear_sdc_wf, func_reports_wf, [
-                        ('outputnode.out_mask_report', 'inputnode.bold_mask_report')]),
-                    (nonlinear_sdc_wf, bold_reg_wf, [
-                        ('outputnode.out_warp', 'inputnode.fieldwarp'),
-                        ('outputnode.out_reference_brain', 'inputnode.ref_bold_brain'),
-                        ('outputnode.out_mask', 'inputnode.ref_bold_mask')]),
-                ])
-            
-            if 'template' in output_spaces:
-                # Apply transforms in 1 shot
-                # Only use uncompressed output if AROMA is to be run
-                bold_mni_trans_wf = init_bold_mni_trans_wf(
-                    template=template,
-                    bold_file_size_gb=bold_file_size_gb,
-                    omp_nthreads=omp_nthreads,
-                    output_grid_ref=output_grid_ref,
-                    use_compression=not (low_mem and use_aroma),
-                    use_fieldwarp=(fmaps is not None or use_syn),
-                    name='bold_mni_trans_wf'
-                )
-
-            workflow.connect([
-                (inputnode, bold_mni_trans_wf, [
-                    ('bold_file', 'inputnode.name_source'),
-                    ('t1_2_mni_forward_transform', 'inputnode.t1_2_mni_forward_transform')]),
-                (bold_hmc_wf, bold_mni_trans_wf, [
-                    ('outputnode.bold_split', 'inputnode.bold_split'),
-                    ('outputnode.xforms', 'inputnode.hmc_xforms')]),
-                (bold_mni_trans_wf, outputnode, [('outputnode.bold_mni', 'bold_mni'),
-                                                 ('outputnode.bold_mask_mni', 'bold_mask_mni')]),
-            ])
-
-            if fmaps:
-                workflow.connect([
-                    (sdc_unwarp_wf, bold_mni_trans_wf, [
-                        ('outputnode.out_warp', 'inputnode.fieldwarp'),
-                        ('outputnode.out_mask', 'inputnode.bold_mask')]),
-                ])
-   
-            elif use_syn:
-                workflow.connect([
-                    (nonlinear_sdc_wf, bold_mni_trans_wf, [
-                        ('outputnode.out_warp', 'inputnode.fieldwarp'),
-                        ('outputnode.out_mask', 'inputnode.bold_mask')]),
-                ])
+    workflow.connect([
+        (inputnode, bold_reference_wf, [('bold_file', 'inputnode.bold_file')]),
+        (bold_reference_wf, bold_hmc_wf, [
+            ('outputnode.raw_ref_image', 'inputnode.raw_ref_image')]),
+        (bold_reference_wf, func_reports_wf, [
+            ('outputnode.validation_report', 'inputnode.validation_report')]),
+        (summary, func_reports_wf, [('out_report', 'inputnode.summary_report')]),
+        ])
```

Comment:
In this file, I honestly was a little overwhelmed by trying to figure out all the branching, so I made some fairly major changes. The primary one is that I switched from checking `if dismiss_t1w`/`else` to `if not dismiss_t1w`. The things that could be left unchanged from the current upstream master I left unchanged, and then stuck the necessary changes in `if not dismiss_t1w`.

The other change is that I made `use_syn = False` if `dismiss_t1w` is `True`. This means that a lot of the logic around susceptibility distortion correction could be left alone.

Reply:
Ok I'll try this version out. I though bold_reference depended on T1 though, so why is it outside the if not dismiss_t1 loop?

Reply:
`bold_reg_wf` depends on T1. `bold_reference_wf` only depends on the incoming BOLD image.